python-sdk/nuscenes/map_expansion/tests/test_arcline_path_utils.py

```diff
@@ -0,0 +1,138 @@
+                           (391.7142849867393, 1100.464077182952, 2.7365754617298705),
+                           (387.35724292592613, 1101.5723176767192, 3.048461127775915),
+                           (382.87033132963325, 1101.2901176788932, -2.922838513357627),
+                           (378.6864775951582, 1099.6447057425564, -2.610952847311582),
+                           (375.20936805976606, 1096.7948422737907, -2.2990671812655377),
+                           (372.7733659833847, 1093.0160135871613, -2.0002085809158623)])
+        np.testing.assert_allclose(answer, discrete_path)
+
+    def test_length_of_lane(self):
+
+        self.assertEqual(arcline_path_utils.length_of_lane(self.straight_lane),
+                         sum(self.straight_path['segment_length']))
+
+        self.assertEqual(arcline_path_utils.length_of_lane(self.right_lane),
+                         sum(self.right_path['segment_length']))
+
+        self.assertEqual(arcline_path_utils.length_of_lane(self.curved_lane),
+                         sum(self.straight_path['segment_length']) + sum(self.left_path['segment_length']))
+
+    def test_project_pose_to_straight_lane(self):
+
+        theta = 2.739593514975998
+        end_pose = 421.2419602954602 + 10 * math.cos(theta), 1087.9127960414617 + 10 * math.sin(theta), theta
+
+        pose, s = arcline_path_utils.project_pose_to_lane(end_pose, self.straight_lane)
+
+        np.testing.assert_allclose(np.array(pose).astype('int'), np.array(end_pose).astype('int'))
+        self.assertTrue(abs(s - 10) <= 0.5)
+
+    def test_project_pose_not_close_to_lane(self):
+
+        pose = 362, 1092, 1.15
+
+        pose_on_lane, s = arcline_path_utils.project_pose_to_lane(pose, self.right_lane)
+        self.assertListEqual(list(pose_on_lane), self.right_path['start_pose'])
+        self.assertEqual(s, 0)
+
+    def test_project_pose_to_curved_lane(self):
+
+        theta = 2.739593514975998
+        end_pose_1 = 421.2419602954602 + 10 * math.cos(theta), 1087.9127960414617 + 10 * math.sin(theta), theta
+
+        end_pose_2 = 381, 1100, -2.76
+
+        pose, s = arcline_path_utils.project_pose_to_lane(end_pose_1, self.curved_lane)
+        np.testing.assert_allclose(np.array(pose).astype('int'), np.array(end_pose_1).astype('int'))
+        self.assertTrue(abs(s - 10) <= 0.5)
+
+        pose_2, s_2 = arcline_path_utils.project_pose_to_lane(end_pose_2, self.curved_lane)
+        np.testing.assert_allclose(np.array(pose_2[:2]).astype('int'), np.array([380, 1100]))
+        self.assertTrue(abs(s_2 - 44) <= 0.5)
+
+    def test_get_curvature_straight_lane(self):
+
+        curvature = arcline_path_utils.get_curvature_at_distance_along_lane(15, self.straight_lane)
+        self.assertEqual(curvature, 0)
+
+    def test_curvature_curved_lane(self):
+
+        curvature = arcline_path_utils.get_curvature_at_distance_along_lane(53, self.curved_lane)
+        self.assertEqual(curvature, 1 / self.left_path['radius'])
+
+
+
+
+
```

Comment:
Too many blank lines

Reply:
Removed the extra lines. Thanks for pointing that out.